src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
So we could not distinguish the devices on one seat? How would that work out with different keyboards and layouts. I still try to understand how xkbcommon would handle that stuff.